libs/langchain-google-cloud-sql-pg/src/vectorStore.ts

```diff
@@ -327,6 +329,83 @@ class PostgresVectorStore extends VectorStore {
 
     return rows;
   }
+
+  async maxMarginalRelevanceSearch(
+    query: string,
+    options?: MaxMarginalRelevanceSearchOptions<this["FilterType"]>,
+    _callbacks?: Callbacks | undefined
+  ): Promise<Document[]> {
+    const vector = await this.embeddings.embedQuery(query);
+    return await this.maxMarginalRelevantSearchByVector(
+      vector,
+      options?.k,
+      options?.fetchK,
+      options?.lambda,
+      options?.filter
+    );
+  }
+
+  async maxMarginalRelevantSearchByVector(
+    vector: number[],
+    k?: number,
+    fetchK?: number,
+    lambdaMult?: number,
+    filter?: this["FilterType"]
+  ): Promise<Document[]> {
+    const docsAndScores =
+      await this.maxMarginalRelevanceSearchWithScoreByVector(
+        vector,
+        k,
+        fetchK,
+        lambdaMult,
+        filter
+      );
+    const docsList: Document[] = [];
+
+    for (const docScore in docsAndScores) {
+      docsList.push(docsAndScores[docScore]);
+    }
+
+    return docsList;
+  }
+
+  async maxMarginalRelevanceSearchWithScoreByVector(vector: number[], k?: number, fetchK?: number, lambdaMult?: number, filter?: this["FilterType"]): Promise<Document[]> {
```

Comment:
This function and maxMarginalRelevantSearchByVector aren't defined by the JS interface so let's delete them for now and move this logic to "maxMarginalRelevanceSearch"